unlock-app/next.config.js

```diff
@@ -10,23 +10,18 @@ const copyFile = promisify(fs.copyFile)
 // TODO renames these: URLs need to be URLs, hosts need to be hosts... etc
 let requiredConfigVariables = {
   unlockEnv: process.env.UNLOCK_ENV || 'dev',
-  httpProvider: process.env.HTTP_PROVIDER || '127.0.0.1',
+  httpProvider: process.env.HTTP_PROVIDER,
   paywallUrl: process.env.PAYWALL_URL,
   paywallScriptUrl: process.env.PAYWALL_SCRIPT_URL,
   readOnlyProvider: process.env.READ_ONLY_PROVIDER,
-  locksmithHost: process.env.LOCKSMITH_URI || 'http://127.0.0.1:8080',
-  unlockAddress:
-    process.env.UNLOCK_ADDRESS || '0x885EF47c3439ADE0CB9b33a4D3c534C99964Db93', // default for CI
+  locksmithHost: process.env.LOCKSMITH_URI,
+  unlockUrl: process.env.UNLOCK_URL,
 }
 
-// If the URL is set in an env variable, use it - otherwise it'll be overridden in config.js
-if (process.env.UNLOCK_URL)
-  requiredConfigVariables.unlockUrl = process.env.UNLOCK_URL
-
+// If any env variable is missing, fail to run, except for dev which can set its own defaults
 Object.keys(requiredConfigVariables).forEach(configVariableName => {
   if (!requiredConfigVariables[configVariableName]) {
-    if (requiredConfigVariables.unlockEnv === 'test') return
-    if (requiredConfigVariables.unlockEnv === 'dev') {
+    if (['dev', 'test'].indexOf(requiredConfigVariables.unlockEnv) > -1) {
```

Comment:
This will add a lot of logging messages in our logs... which means we need to make sure these vars are set!